OracleAnalytics/dockerfiles/6.4/createAndStartDomain.sh

```diff
@@ -0,0 +1,219 @@
+#  and not use 'as sysdba', c) as per b, but use the Connections API to recover a schema password from the domain (this makes
+#  more sense than (b).
+
+# If a wait timeout is set, block until the DB is available or the timeout is reached.
+# This is useful when starting DB+BI via docker compose.
+DB_WAIT_TIMEOUT=${DB_WAIT_TIMEOUT:-0}
+if (( DB_WAIT_TIMEOUT > 0 )); then
+ if [ -z "$DB_USERNAME" ] || [ -z "$DB_PASSWORD" ]; then
+   echo "DB_USERNAME and DB_PASSWORD must be set if DB_WAIT_TIMEOUT is non-zero"
+   exit 3
+ fi
+ set +e
+ echo "$DB_PASSWORD" | timeout "$DB_WAIT_TIMEOUT" /u01/wait_for_db.sh "$ORACLE_HOME" "$DB_USERNAME" "$DB_HOST" "$DB_PORT" "$DB_SERVICE"
+ exit_code="$?"
+ if [ ! "${exit_code}" -eq 0 ]; then
+    echo "Reached timeout waiting for DB to start - exiting"
+    exit 1
+ fi
+ set -e
+fi
+
+# Create Domain only if 1st execution
+if [ ! -f "$domainCheckFile" ]; then
+  masterResponseFile=$ORACLE_HOME/bi/modules/oracle.bi.configassistant/response.txt
+
+  tempDir=$(mktemp -dt "create_domain_XXXXXXXXXX")
+  trap 'rm -rf -- "$tempDir"; kill -TERM $PID' INT TERM
+
+  responseFile=$tempDir/response.txt
+  cp "$masterResponseFile" "$responseFile"
+
+  replacements="  -e \"s|@@DOMAIN_TYPE@@|DOMAIN_TYPE_EXPANDED|g\""
+  replacements+=" -e \"s|@@CONFIGURE_ESSBASE@@|true|g\""
+  replacements+=" -e \"s|@@CONFIGURE_BIEE@@|true|g\""
+  replacements+=" -e \"s|@@CONFIGURE_BIP@@|true|g\""
+  replacements+=" -e \"s|@@DOMAIN_NAME@@|$DOMAIN_NAME|g\""
+  replacements+=" -e \"s|@@DOMAINS_DIR@@|$DOMAINS_DIR|g\""
+  replacements+=" -e \"s|@@ADMIN_USER_NAME@@|$ADMIN_USERNAME|g\""
+  replacements+=" -e \"s|@@ADMIN_PASSWORD@@|$ADMIN_PASSWORD|g\""
+  replacements+=" -e \"s|@@SCHEMA_TYPE@@|SCHEMA_TYPE_NEW|g\""
+  replacements+=" -e \"s|@@DB_TYPE@@|ORACLE|g\""
+  replacements+=" -e \"s|@@DB_CONNECT_STRING@@|$DB_HOST:$DB_PORT:$DB_SERVICE|g\""
+  replacements+=" -e \"s|@@DB_ADMIN_USERNAME@@|$DB_USERNAME|g\""
+  replacements+=" -e \"s|@@DB_PASSWORD@@|$DB_PASSWORD|g\""
+  replacements+=" -e \"s|@@PREFIX@@|$SCHEMA_PREFIX|g\""
+  replacements+=" -e \"s|@@NEW_DB_SCHEMA_PASSWORD@@|$SCHEMA_PASSWORD|g\""
+  replacements+=" -e \"s|@@NEW_DB_SCHEMA_PASSWORD@@|$SCHEMA_PASSWORD|g\""
+  replacements+=" -e \"s|@@APPLICATION_TYPE@@|APPLICATION_TYPE_EMPTY|g\""
+  replacements+=" -e \"s|@@AUTO_DROP_SCHEMAS@@|false|g\""
+  replacements+=" -e \"s|@@DEFAULT_SI_KEY@@|ssi|g\""
+  replacements+=" -e \"s|@@PORT_RANGE_START@@|9500|g\""
+  replacements+=" -e \"s|@@PORT_RANGE_END@@|9999|g\""
+
+  eval sed -i "$replacements" "$responseFile"
+
+  # TODO (minor as scale-out not supported) - for SDD setting, need to not start the stack, but mv sdd and mod the xml (and manually dump diag zip?)
+  # TODO (minor) - hostname can change after image update, could realign nodemanager settings.
+  # TODO (minor) - autoport is in use, but the container must expose fixed ports.  This works by chance.
```

Comment:
Either do these things or remove them from this file. Do not leave TODO items in published files